src/tests/ftest/util/mdtest_utils.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+
+from __future__ import print_function
+import os
+import subprocess
+import json
+
+class MdtestFailed(Exception):
+    """Raise if mdtest failed"""
+
+#pylint: disable=R0903
+class MdtestParam(object):
+    """
+    Defines a object representing a single mdtest command line parameter.
+    """
+
+    def __init__(self, str_format, default=None):
+        """Create a MdtestParam object.
+        Args:
+            str_format (str): format string used to convert the value into an
+                mdtest command line argument string
+            default (object): default value for the param
+        """
+        self.str_format = str_format
+        self.default = default
+        self.value = default
+
+    def __str__(self):
+        """Return a MdtestParam object as a string.
+        Returns:
+            str: if defined, the Mdtest parameter, otherwise an empty string
+        """
+        if isinstance(self.default, bool) and self.value:
+            return self.str_format
+        elif not isinstance(self.default, bool) and self.value:
+            return self.str_format.format(self.value)
+        else:
+            return ""
+
+    def set_yaml_value(self, name, test, path="/run/mdtest/*"):
+        """Set the value of the parameter using the test's yaml file value.
+        Args:
+            name (str): name associated with the value in the yaml file
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace. Defaults to "/run/mdtest/*"
+        """
+        self.value = test.params.get(name, path, self.default)
+
+
+class MdtestCommand(object):
```

Comment:
I'd suggest inheriting `CommandWithParameters` from `src/tests/ftest/util/command_utils.py` in #702